### PR TITLE
Issue/2919 datetime aware

### LINKED ISSUE #2896: Have server report all times in UTC
- Update the json serializer to produce UTC timestamps 
Create follow up ticket for updating the database

### LINKED ISSUE #2919: make all datetime data and code timezone-aware
#2896 was about making sure the server methods respond with times in UTC. This ticket expands on that by making sure all of core is timezone-aware.

- database: include timezone information in database entires (`WITH TIME ZONE`).
- code: make sure all code uses timezone-aware datetime instances. Timezone-aware instances are not compatible with naive instances.
- API boundary: input via the API should support ISO timezone suffixes. Return values from methods should either return times in UTC without timezone information (as it does now) or add ISO timezone suffixes as well, depending on what is supported by API clients at the time of implementation.

### PR DESCRIPTION
**Pull request by sanderr**
# Description

- Makes everything internal timezone-aware
- API input accepts both timezone-aware and naive timestamps. Naive timestamps are assumed to be in UTC. Same for Python client input.
- API output returns naive timestamps in UTC.

closes #2919
closes #2896

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] ~~End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

**Review by bartv — APPROVED** [review]

**Review by wouterdb — APPROVED** [review]

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: src/inmanta/server/services/paramservice.py [hunk 1]
```diff
@@ -264,6 +264,10 @@ async def list_params(self, env: data.Environment, query: Dict[str, str]) -> Api
             {
                 "parameters": params,
                 "expire": self._fact_expire,
-                "now": datetime.datetime.now().isoformat(timespec="microseconds"),
+                # Return datetime in UTC without explicit timezone offset
+                "now": datetime.datetime.now()
+                .astimezone(datetime.timezone.utc)
+                .replace(tzinfo=None)
+                .isoformat(timespec="microseconds"),
```
[the remark this answers is not in the document]
Comment (sanderr) [reply src/inmanta/server/services/paramservice.py:271]: That's a good suggestion, I'll see what abstractions I can make.

[the remark this answers is not in the document]
Comment (sanderr) [reply src/inmanta/server/services/paramservice.py]: Is it better now?

File: src/inmanta/server/agentmanager.py [hunk 2]
```diff
@@ -729,7 +733,10 @@ async def list_agents(
             **query,
         )
 
-        return 200, {"agents": [a.to_dict() for a in ags], "servertime": datetime.now().isoformat(timespec="microseconds")}
+        return 200, {
+            "agents": [a.to_dict() for a in ags],
+            "servertime": datetime.now().astimezone(timezone.utc).replace(tzinfo=None).isoformat(timespec="microseconds"),
```
Comment (arnaudsjs) [inline src/inmanta/server/agentmanager.py:738]:
```suggestion
            "servertime": datetime_utc_isoformat(datetime.now())
```

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: src/inmanta/server/agentmanager.py (hunk 2 above)
Comment (sanderr) [reply src/inmanta/server/agentmanager.py]: Thanks, I missed that one.

**Review by arnaudsjs — APPROVED** [review]

**Pull request closed without merging**